### ISSUE TITLE
Release automation

### ISSUE THREAD

**Issue by luxaritas** [eternagame/workspace-helpers#55]
semantic-release doesn't support monorepos and I'm not super excited about any of the forks, however https://github.com/googleapis/release-please looks like a good option!

I think it should make sense to add a generator for this to our nx plugin (which we should use in this repo!)

**Comment by luxaritas** [comment]
release-please has some idiosyncrasies around things like handling releasing packages when we update our root package.json or package-lock.json, cutting out the scope from package names, etc. Apparently lerna can generate changelogs based on conventional commits, so I think that's worth looking into. It would be nice to have release PRs still - I'll need to look into what that would take

**Comment by luxaritas** [comment]
Lerna can also do "canary releases" which could be super helpful for testing (and we could theoretically do that in a github workflow that runs on pushes to main)

**Comment by luxaritas** [comment]
I think we should prompt in our preset the intended release workflow. Basically there are a few things we want to kow:
- Should we set up the release workflow at all?
- Should publishing be set up in the release workflow?
- Should packages be public or private by default when generated?

For the last item, we can set a [generator preset](https://nx.dev/configuration/packagejson#generators) for our package generator (if we needed to do any further customization there for releaseable packages, we could)
